packages/style-engine/class-wp-style-engine.php

```diff
@@ -605,9 +610,15 @@ protected static function get_url_or_value_css_declaration( $style_value, $style
 
 			if ( isset( $style_definition['property_keys']['default'] ) ) {
 				$value = null;
-
-				if ( ! empty( $style_value['url'] ) ) {
-					$value = "url('" . $style_value['url'] . "')";
+				if ( ! empty( $style_value['url'] ) && is_string( $style_value['url'] ) ) {
+					$url = $style_value['url'];
+					if ( ! empty( $options['base_url'] ) ) {
```

Comment:
So the idea here is that if you pass a base url to the style engine it will append it to all URL-like values.

Trying to work out if this is a bad idea and whether the url should come already edited.

Pros:

- A quick way to create absolute URLs for multiple values
- Centralized


Cons:

- It shouldn't interfere with other URLs, e.g., `url(data:image/png;base64,iRxVB0…)`, `url(myFont.woff)`, `url(#IDofSVGpath)`; Maybe we could also check for `$style_value['source'] === "theme"` to get around that.

Reply:
Neat idea to centralize! Some quick thoughts off the top of my head:

* If we're unsure of the long-term shape of anything, place the logic in the consumer of the style engine, rather than in the style engine itself. (I.e. in this case, in `background.php`).
* If we're sure of the shape, then centralizing is a good idea. In this case, I'd lean toward calling it something like `'theme_base_url'` and only prepending if `$style_value['source'] === "theme"` as you mention.

Reply:
Thanks! I was about to ping you 😄 

> If we're unsure of the long-term shape of anything, place the logic in the consumer of the style engine, rather than in the style engine itself. (I.e. in this case, in background.php).

That's a great summary.

Are we ever sure of anything? 😄 

This was mainly an experiment to see if reduced the code (a bit) and worked (yes).

Considering how new background supports is, I might err on the side of caution and roll back to what we had. It's not pretty, but it's more flexible.

Reply:
> Are we ever sure of anything? 😄

In Gutenberg? Great question 😆

> Considering how new background supports is, I might err on the side of caution and roll back to what we had. It's not pretty, but it's more flexible.

Sounds good! And you've got a solid plan for a follow-up PR if we decide we'd like to centralise 👍